File: backend/app/schemas/tutor.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any

class TutorChatRequest(BaseModel):
    message: str
    context: Optional[Dict[str, Any]] = None

class TutorChatResponse(BaseModel):
    message_id: str
    role: str = "ai"
    text: str
    timestamp: str
    context_used: Optional[Dict[str, Any]] = None

class ChatHistoryItem(BaseModel):
    id: str
    sender: str
    text: str
    timestamp: str

File: src/api/tutor.ts
import { apiClient } from './client';

export interface ChatMessage {
  id: string;
  sender: 'user' | 'ai';
  text: string;
  timestamp: string;
}

export interface TutorResponse {
  message_id: string;
  role: string;
  text: string;
  timestamp: string;
  context_used?: any;
}

export const tutorApi = {
  sendMessage: async (message: string, context?: any): Promise<TutorResponse> => {
    return apiClient<TutorResponse>('/tutor/chat', {
      method: 'POST',
      body: JSON.stringify({ message, context }),
    });
  },

  getHistory: async (): Promise<ChatMessage[]> => {
    return apiClient<ChatMessage[]>('/tutor/history');
  },
};
